dcp-0006/dcp-0006.mediawiki

```diff
@@ -0,0 +1,345 @@
+|-
+|TxOut[0]||[0]OP_RETURN<br>[1]OP_DATA_32||Bytes that make the entire transaction unique. First 8 bytes are the little endian encoded value from TxIn[0]. The following 24 bytes are random.
+|-
+|TxOut[1:N]||[0]OP_TGEN<br>[1:M]P2PKH or P2SH script||These are the individual payouts from the treasury account.
+|}
+
+Additional consensus enforced characteristics:
+* The previous out point for TxIn[0] shall be null.
+* The TSpend transaction fraud proof shall be null.
+* The length of the signature script in TxIn[0] shall be 100 bytes ([OP_DATA_64] [signature] [OP_DATA_33] [PiKey] [OP_TSPEND] = 1 + 64 + 1 + 33 + 1 = 100)
+* A TSpend transaction shall not be allowed in a block that is not a TVI.
+* A TSpend transaction shall not be allowed in a block if the start of the voting window is prior to stake validation height.
+* A TSpend transaction shall not be allowed in a block if the Expiry is outside of the valid TSpend validity window.
+* The value of input 0 of a TSpend transaction shall be int64 little endian encoded in the output 0 script in bytes 2 through 10. The value of bot shall be identical.
+* A TSpend transaction shall not be allowed in a block if it has been mined in a prior block (this is to enable support for short-circuiting a vote) that is in the chain of ancestors.
+* A TSpend transaction shall not be allowed in a block if it does not reach quorum which is calculated as follows: (sum(yes votes) + sum(no votes)) * TVQM / TVQD
+* A TSpend transaction shall not be allowed in a block if it does not reach enough yes votes  which is calculated as follows: (sum(votes cast) + sum(possible remaining votes)) * TVRM / TVRD
+* A TSpend transaction shall not deplete the entire treasury. The treasury balance shall not become negative.
+* A TSpend transaction shall not spend more than 150% of avg(sum(tspends), TPE).
+* A TSpend transaction shall not be allowed in a block if the Schnorr public key is not well-known (encoded in chain parameters).
+* A TSpend transaction shall not be allowed in a block if the Schnorr signature of the transaction does not match the provided well-known public key.
+
+===Treasury Account Balance===
+
+All treasury funds in or out are subject to coinbase maturity rules and thus
+the current treasury account balance is determined by adding the sum of all
+additions and subtractions from the treasury account as of coinbase maturity
+blocks ago.
+
+The treasury account balance shall remain 0 until the treasury agenda is voted
+in and the block height reaches activation height + coinbase maturity. Only at
+that point are are any additions and subtractions possible.
+
+===Mempool===
+
+In order to allow TSpend transactions into the mempool some modifications have to
+be made.
+
+* A treasurybase transaction shall not be allowed in the mempool.
+* The mempool shall not allow more than 7 TSpend concurrent transactions in the mempool at the same time.
```

Comment:
"concurrent" seems redundant here

Reply:
I feel it makes it more clear. Anyone disagrees?